rs/ic_os/config/src/lib.rs

```diff
@@ -185,59 +154,270 @@ pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettin
+    }
+
+    #[test]
+    fn test_is_valid_ipv6_prefix() {
+        assert!(is_valid_ipv6_prefix("2a00:1111:1111:1111"));
+        assert!(is_valid_ipv6_prefix("2a00:111:11:11"));
+        assert!(is_valid_ipv6_prefix("2602:fb2b:100:10"));
+    }
+
+    #[test]
+    fn test_parse_config_line() {
+        assert_eq!(
+            parse_config_line("key=value"),
+            Some(("key".to_string(), "value".to_string()))
+        );
+        assert_eq!(
+            parse_config_line("   key   =   value   "),
+            Some(("key".to_string(), "value".to_string()))
+        );
+        assert_eq!(parse_config_line(""), None);
+        assert_eq!(parse_config_line("# this is a comment"), None);
+        assert_eq!(parse_config_line("keywithoutvalue"), None);
+        assert_eq!(
+            parse_config_line("key=value=extra"),
+            Some(("key".to_string(), "value=extra".to_string()))
+        );
+    }
+
+    #[test]
+    fn test_config_map_from_path() -> Result<()> {
+        let mut temp_file = NamedTempFile::new()?;
+        let file_path = temp_file.path().to_path_buf();
+
+        writeln!(temp_file, "key1=value1")?;
+        writeln!(temp_file, "key2=value2")?;
+        writeln!(temp_file, "# This is a comment")?;
+        writeln!(temp_file, "key3=value3")?;
+        writeln!(temp_file)?;
+
+        let config_map = config_map_from_path(&file_path)?;
+
+        assert_eq!(config_map.get("key1"), Some(&"value1".to_string()));
+        assert_eq!(config_map.get("key2"), Some(&"value2".to_string()));
+        assert_eq!(config_map.get("key3"), Some(&"value3".to_string()));
+        assert_eq!(config_map.get("bad_key"), None);
+
+        // CRLF test
+        let mut temp_file_crlf = NamedTempFile::new()?;
+        let file_path_crlf = temp_file_crlf.path().to_path_buf();
+
+        writeln!(temp_file_crlf, "key4=value4\r\nkey5=value5\r\n")?;
+
+        let config_map_crlf = config_map_from_path(&file_path_crlf)?;
+
+        assert_eq!(config_map_crlf.get("key4"), Some(&"value4".to_string()));
+        assert_eq!(config_map_crlf.get("key5"), Some(&"value5".to_string()));
+        assert_eq!(config_map.get("bad_key"), None);
+
+        std::fs::remove_file(&file_path)?;
+        std::fs::remove_file(&file_path_crlf)?;
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_get_config_ini_settings() -> Result<()> {
+        // Test valid config.ini
+        let mut temp_file = NamedTempFile::new()?;
+        writeln!(temp_file, "ipv6_prefix=2a00:fb01:400:200")?;
+        writeln!(temp_file, "ipv6_address=2a00:fb01:400:200::/64")?;
+        writeln!(temp_file, "ipv6_gateway=2a00:fb01:400:200::1")?;
+        writeln!(temp_file, "ipv4_address=212.71.124.178")?;
+        writeln!(temp_file, "ipv4_gateway=212.71.124.177")?;
+        writeln!(temp_file, "ipv4_prefix_length=28")?;
+        writeln!(temp_file, "domain=example.com")?;
+        writeln!(temp_file, "verbose=false")?;
+
+        let temp_file_path = temp_file.path().to_path_buf();
```

Comment:
`temp_file.path()` should be enough (same below)

Reply:
Removed!